### PR TITLE
feat: add integration tests for ES Search DAO.

### PR DESCRIPTION
**Pull request by jplaisted**
Note the existing test model (BarSaerchDocument) is a little too vague and only had one field. Rather than expand it, I figured I'd make a more congrete and silly piece of test data for Pizzas.

Note: tests are the same across 5 and 7. Which makes sense as the DAO API did not change across versions.

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Review by woodyzhou — APPROVED** [review]
LGTM

File: testing/elasticsearch-dao-integ-testing-docker-7/src/test/java/com/linkedin/metadata/testing/ESSearchDaoIntegTest.java [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+    _bulkDao.upsertDocument(
+        new PizzaSearchDocument().setSize(PizzaSize.MEDIUM).setToppings(new StringArray("Sausage Links")).setUrn(urn0),
+        urn0.toString());
+    _bulkDao.upsertDocument(
+        new PizzaSearchDocument().setSize(PizzaSize.LARGE).setToppings(new StringArray("Sausage crumble")).setUrn(urn1),
+        urn1.toString());
+    _searchIndex.getRequestContainer().flushAndSettle();
+
+    // when
+    final AutoCompleteResult result = _searchDao.autoComplete("Sausage", null, null, 10);
```
Comment (camelliazhang) [inline testing/elasticsearch-dao-integ-testing-docker-7/src/test/java/com/linkedin/metadata/testing/ESSearchDaoIntegTest.java:116]: Might also test autoComplete("Sau")?

File: testing/elasticsearch-dao-integ-testing-docker-7/src/test/java/com/linkedin/metadata/testing/ESSearchDaoIntegTest.java [hunk 2]
```diff
@@ -0,0 +1,439 @@
+package com.linkedin.metadata.testing;
```
Comment (camelliazhang) [inline testing/elasticsearch-dao-integ-testing-docker-7/src/test/java/com/linkedin/metadata/testing/ESSearchDaoIntegTest.java:1]: Add test for aggregations/facets? It's supported in SearchDAO.

**Review by camelliazhang — APPROVED** [review]
Added comments, otherwise, LGTM. Good job!

File: testing/elasticsearch-dao-integ-testing-docker-7/src/test/java/com/linkedin/metadata/testing/ESSearchDaoIntegTest.java (hunk 1 above)
Comment (jplaisted) [reply testing/elasticsearch-dao-integ-testing-docker-7/src/test/java/com/linkedin/metadata/testing/ESSearchDaoIntegTest.java:116]: Done

File: testing/elasticsearch-dao-integ-testing-docker-7/src/test/java/com/linkedin/metadata/testing/ESSearchDaoIntegTest.java (hunk 2 above)
Comment (jplaisted) [reply testing/elasticsearch-dao-integ-testing-docker-7/src/test/java/com/linkedin/metadata/testing/ESSearchDaoIntegTest.java:1]: I don't know what those are! Maybe lets do it in a follow up. I can learn what they are, but we don't have to block these on that.

Comment (camelliazhang) [reply testing/elasticsearch-dao-integ-testing-docker-7/src/test/java/com/linkedin/metadata/testing/ESSearchDaoIntegTest.java:1]: SG (Approved)

**Pull request merged**